### DISCUSSION TITLE
conn.interrupt() in duckDB python API

### DISCUSSION THREAD

**Discussion by lowfi117** [duckdb/duckdb#5937] [General]
Is there a python interrupt API in duckDB similar to sqlite3 conn.interrupt()? I have long queries and want to stop the old query while new query come in?

**Comment by Mytherin** [comment]
I don't think this is there yet in the Python client but we support it in the C++ client. It should be rather easy to add. I will open a feature request.

**Reply by lowfi117** [reply]
Awesome, thanks! This will provide great flexibility when dealing with long queries.

**Reply by lowfi117** [reply]
@Mytherin Before the `interrupt()` function is available, is there any other way to exit the long query from python API? I saw in one post the query could be aborted from `interrupt_main()`, but I don't want to shutdown the whole app. Any idea?

**Reply by Mytherin** [reply]
DuckDB should respect regular Python interrupts (triggered by e.g. Ctrl+C). I would have a look to see if you can interrupt regular Python operations on a different thread - that should work with DuckDB as well.
